packages/schematics/angular/migrations/migration-collection.json

```diff
@@ -14,6 +14,12 @@
       "version": "17.0.0",
       "factory": "./update-17/update-workspace-config",
       "description": "Replace deprecated options in 'angular.json'."
+    },
+    "use-application-builder": {
+      "version": "17.1.0",
+      "factory": "./update-17/use-application-builder",
+      "description": "Update application projects to use '@angular-devkit/build-angular:application' builder.",
+      "optional": true
```

Comment:
After our discussion today, I think the `outputPath` change is the real issue here. As is today, I think we need `manual` because this change will break any existing deployment pipelines. If we're able to apply this migration without changing the output path, then I think `optional` would be viable.